integration-tests/jsonpath/src/main/java/org/apache/camel/quarkus/component/json/path/it/JsonPathResource.java

```diff
@@ -68,7 +84,60 @@ public String getFullName(String personRequestJson) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.TEXT_PLAIN)
     public String getAllCarColors(String carsRequestJson) {
-        LOG.infof("Getting all car colors from json cars request: %s", carsRequestJson);
+        LOG.debugf("Getting all car colors from json cars request: %s", carsRequestJson);
         return producerTemplate.requestBody("direct:getAllCarColors", carsRequestJson, String.class);
     }
+
+    @Path("/splitBooksShouldReturnTwoPrices")
+    @GET
+    public void splitBooksShouldReturnThreePrices() throws InterruptedException {
+        LOG.debugf("Calling splitBooksShouldReturnThreePrices()");
+
+        String body = "{\"books\": [{\"price\": 30},{ \"price\": 20}]}";
+        MockEndpoint mockPrices = context.getEndpoint("mock:prices", MockEndpoint.class);
+        mockPrices.expectedMessageCount(2);
+        producerTemplate.requestBody("direct:splitBooks", body, String.class);
+        mockPrices.assertIsSatisfied();
+
+        List<Exchange> exchanges = mockPrices.getReceivedExchanges();
+        assertNotNull(exchanges);
+        assertEquals(2, exchanges.size());
+
+        Map<?, ?> firstRow = exchanges.get(0).getIn().getBody(Map.class);
+        assertNotNull(firstRow);
+        assertEquals(30, firstRow.get("price"));
+
+        Map<?, ?> secondRow = exchanges.get(1).getIn().getBody(Map.class);
+        assertNotNull(secondRow);
+        assertEquals(20, secondRow.get("price"));
+    }
+
+    @Path("/setHeaderWithJsonPathExpressionEvaluatingAnotherHeaderShouldSucceed")
+    @GET
+    public void setHeaderWithJsonPathExpressionEvaluatingAnotherHeaderShouldSucceed() throws InterruptedException {
+        LOG.debugf("Calling setHeaderWithJsonPathExpressionEvaluatingAnotherHeaderShouldSucceed()");
+
+        String json = "{\"book\": {\"price\": 25} }";
+        MockEndpoint mockSetHeader = context.getEndpoint("mock:setHeader", MockEndpoint.class);
+        mockSetHeader.expectedMessageCount(1);
+        mockSetHeader.expectedHeaderReceived("price", 25);
+        producerTemplate.requestBodyAndHeader("direct:setHeader", null, "jsonBookHeader", json, Message.class);
+        mockSetHeader.assertIsSatisfied();
+    }
+
+    @Path("/getAuthorsFromJsonStream")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_OCTET_STREAM)
+    public List<?> getAuthorsFromJsonStream(byte[] jsonBytes, @QueryParam("encoding") String encoding) {
+        LOG.debugf("Getting authors from JsonStream with encoding '%s' and %d bytes", encoding, jsonBytes.length);
+
+        ByteArrayInputStream jsonStream = new ByteArrayInputStream(jsonBytes);
```

Comment:
Maybe wrap in a `try` block?

Reply:
I've pushed a commit more with try/autocloseable. Let me know.